### PR TITLE
[ENG-1524] use envs from eas.json when resolving app config

### PR DESCRIPTION
**Pull request by dsokal**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.
- [ ] I've tagged the changelog entry with `[EAS BUILD API]` if it's a part of a breaking change to EAS Build API (only possible when updating `@expo/eas-build-job` package).

# Why

https://linear.app/expo/issue/ENG-1524/use-env-field-in-easjson-to-evaluate-appconfigjs
We want to use the env vars from `eas.json` when evaluating `app.config.js`.

# How

- App config is now read separately for Android and iOS - this is because those platforms are using separate build profiles, and they can have different sets of env vars set.
- I got rid of the `commandCtx`. All properties have been merged into `buildCtx`.

# Test Plan

- I ran a couple builds locally (both with `-p PLATFORM` flag and `-p all`) and haven't noticed any regression.
- I had a sample project with the following `app.config.js`:
```js
let version;
if (process.env.BLAH === '123') {
  version = '1.0.1';
} else if (process.env.BLAH === '456') {
  version = '1.0.2';
} else {
  version = '1.0.0';
}

const config = {
  "expo": {
    "name": "blahblah",
    "slug": "blahblah",
    "version": version,
    ...
  }
};

module.exports = config
```
  I ran two builds (one for iOS and another for Android). The builds had different versions set.

### CODE REVIEW

File: packages/eas-cli/src/project/projectUtils.ts [hunk 1]
```diff
@@ -111,7 +110,10 @@ export async function setProjectIdAsync(projectDir: string): Promise<ExpoConfig
   return (result.config as AppJSONConfig)?.expo;
 }
 
-export async function getProjectIdAsync(exp: ExpoConfig): Promise<string> {
+export async function getProjectIdAsync(
+  exp: ExpoConfig,
+  options: { env?: Env } = {}
```
Comment (wkozyra95) [inline packages/eas-cli/src/project/projectUtils.ts:115]: ?
you are not using that field

File: packages/eas-cli/src/error/ExitError.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class ExitError extends Error {
```
Comment (wkozyra95) [inline packages/eas-cli/src/error/ExitError.ts:1]: check package `@oclif/errors`

File: packages/eas-cli/src/credentials/context.ts [hunk 3]
```diff
@@ -45,12 +46,14 @@ export async function createCredentialsContextAsync(
...
-  let expoConfig: ExpoConfig | undefined;
-  try {
-    const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
-    expoConfig = exp;
-  } catch (error) {
-    // ignore error, context might be created outside of expo project
+  let expoConfig: ExpoConfig | undefined = options.exp;
+  if (!expoConfig) {
+    try {
+      const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
```
Comment (wkozyra95) [inline packages/eas-cli/src/credentials/context.ts:52]: consider using `getExpoConfig` here

File: packages/eas-cli/src/build/ios/credentials.ts [hunk 4]
```diff
@@ -16,20 +16,16 @@ export async function ensureIosCredentialsAsync(
     return;
   }
 
-  const { projectDir } = buildCtx.commandCtx;
-
-  const ctx = await createCredentialsContextAsync(projectDir, {
-    nonInteractive: buildCtx.commandCtx.nonInteractive,
+  const ctx = await createCredentialsContextAsync(buildCtx.projectDir, {
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/ios/credentials.ts:19]: nit pick: for code in build directory we should use variable name `ctx` for build context and for contexts value from different scopes we should use name with some prefix e.g. credentialsCtx

File: packages/eas-cli/src/project/expoConfig.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    for (const [key, value] of Object.entries(env)) {
+      process.env[key] = value;
+    }
+    const { exp } = getConfig(projectDir, {
+      skipSDKVersionRequirement: true,
+      isPublicConfig: true,
+    });
+    return exp;
+  } finally {
+    process.env = originalProcessEnv;
```
Comment (wkozyra95) [inline packages/eas-cli/src/project/expoConfig.ts:16]:
```suggestion
  const originalProcessEnv: NodeJS.ProcessEnv = process.env;
  try {
    process.env = {
      ...process.env,
      ...env,
    };
    const { exp } = getConfig(projectDir, {
      skipSDKVersionRequirement: true,
      isPublicConfig: true,
    });
    return exp;
  } finally {
    process.env = originalProcessEnv;
```

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/project/projectUtils.ts (hunk 1 above)
Comment (dsokal) [reply packages/eas-cli/src/project/projectUtils.ts:115]: https://github.com/expo/eas-cli/pull/532/files#diff-e6ddd74b196c39db9302b9ec339e5eed15a1d6675d6b39f06898438e29feac1dR136

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**